### PR TITLE
ouroboros-network-api: make SizeInBytes a newtype

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

File: ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Node/Serialisation.hs [hunk 1]
```diff
@@ -75,7 +75,7 @@ instance Serialise ext => SerialiseNodeToNodeConstraints (MockBlock ext) where
       7 {- CBOR-in-CBOR -} + 1 {- encodeListLen 2 -} + hdrSize + bodySize
     where
       hdrSize  = fromIntegral (Lazy.length (serialise hdr))
-      bodySize = simpleBodySize (simpleHeaderStd hdr)
+      bodySize = fromIntegral (simpleBodySize (simpleHeaderStd hdr))
```
Comment (dnadales) [inline ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Node/Serialisation.hs:78]: Why do we need `fromIntegral` here?

File: ouroboros-network-api/src/Ouroboros/Network/SizeInBytes.hs [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
 import           Data.Word (Word32)
 
-type SizeInBytes = Word32
+import           NoThunks.Class (NoThunks (..))
+
+newtype SizeInBytes = SizeInBytes { getSizeInBytes :: Word32 }
+  deriving (Show, Eq, Ord)
+  deriving Enum     via Word32
+  deriving Num      via Word32
+  deriving Real     via Word32
```
Comment (dnadales) [inline ouroboros-network-api/src/Ouroboros/Network/SizeInBytes.hs:13]: Do we need this instance? It seems strange that `SizeInBytes` is an instance of `Real`.

**Review by dnadales — COMMENTED** [review]
Thanks! Only have one question:

What prevents us from addressing these TODO's is this PR?

File: ouroboros-network-api/src/Ouroboros/Network/SizeInBytes.hs (hunk 2 above)
Comment (coot) [reply ouroboros-network-api/src/Ouroboros/Network/SizeInBytes.hs:13]: `Real` just gives access to `toRational :: SizeInBytes -> Rational` which is useful when one needs to do some division.

File: ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Node/Serialisation.hs (hunk 1 above)
Comment (coot) [reply ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Node/Serialisation.hs:78]: `estimateBlockSize` supposed to create `SizeInBytes` but `simpleBodySize` gives `Word32`.

**Comment by coot** [comment]
> What prevents us from addressing these TODO's is this PR?

Because that's consensus internals, but feel free to fix it :smile:.

**Comment by coot** [comment]
The TODO's introduced in this PR are recorded in #4216.

**Pull request merged**